src/back/auth/src/service/service.cpp

```diff
@@ -252,4 +256,38 @@ void Service::updateTokens(model::Session& session) const
 	session._idToken = tokens._idToken;
 }
 
+model::UserInfo Service::GetUserInfoById(const std::string& id, const std::string& sessionId)
+{
+	// Получаем сессию из базы
+	auto session = _session.Table().Get(sessionId);
+
+	// Обновляем OIDC токены если требуется
+	if (_tokenizer.IsExpired(session._accessToken))
+	{
+		// TODO: dest lock!
+		updateTokens(session);
+		_session.Table().UpdateTokens(session);
+	}
+
+	// Запрашиваем у OIDC инфу о пользователе
+	return _oidc.GetUserInfoById(id, session._accessToken);
+}
```

Comment:
Давай вынесем кусок получения сессии во что то вроде getFrashSession. И в функции GetUserInfo тоже.
```suggestion
model::Session Service::GetUserInfoById(const std::string& id, const std::string& sessionId)
{
	// Получаем сессию из базы
	auto session = _session.Table().Get(sessionId);

	// Обновляем OIDC токены если требуется
	if (_tokenizer.IsExpired(session._accessToken))
	{
		// TODO: dest lock!
		updateTokens(session);
		_session.Table().UpdateTokens(session);
	}
	return session;
}

model::UserInfo Service::GetUserInfoById(const std::string& id, const std::string& sessionId)
{
	auto session = getFrashSession(sessionId);
	return _oidc.GetUserInfoById(id, session._accessToken);
}
```

Reply:
Готово.